cmd/ctr/commands/images/mount.go

```diff
@@ -130,14 +137,21 @@ When you are done, use the unmount command.
 			}
 		}
 
-		if err := mount.All(mounts, target); err != nil {
-			if err := s.Remove(ctx, target); err != nil && !errdefs.IsNotFound(err) {
-				fmt.Fprintln(context.App.ErrWriter, "Error cleaning up snapshot after mount error:", err)
+		if context.Bool("show") {
+			commands.PrintAsJSON(mounts)
+		}
+
+		if !context.Bool("dry-run") {
```

Comment:
Now if this command is run with a `dry-run` option we will still create a snapshot for it and that snapshot will continue to exist. I think we probably should remove the snapshot before exiting if `dry-run` is enabled.

Reply:
That's tricky because for our use case we don't actually want to mount or remove the snapshot since we intend to use it in our functional tests. Perhaps yet another flag could be added? :p

Reply:
Having a dry-run that has left-over side-effects doesn't seem right, though 😟